pkg/ipam/metrics/metrics.go

```diff
@@ -161,6 +162,16 @@ func NewPrometheusMetrics(namespace string, registry metrics.RegisterGatherer) *
 		),
 	}, []string{"type", "status", "subnet_id"})
 
+	m.BackgroundSyncDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
```

Comment:
What about using an HistogramVec to differentiate with labels successful resync from failed resync?

Reply:
Good point! Added status label, please take another look